qt-models/diveplannermodel.cpp

```diff
@@ -984,7 +984,7 @@ struct divedatapoint * DivePlannerPointsModel::cloneDiveplan(struct diveplan *pl
 	return last_segment;
 }
 
-int DivePlannerPointsModel::analyzeVariations(struct decostop *min, struct decostop *mid, struct decostop *max, const char *unit)
+int DivePlannerPointsModel::analyzeVariations(struct decostop *min, struct decostop *mid, struct decostop *max, QString unit)
```

Comment:
Why this change? What is the advantage to go through the "fatter" class QString instead of *char?

Reply:
By the way: is there any agreement on whether to pass by const reference (`const QString &unit`) or by value (`QString unit`)? It doesn't matter too much, since `QString`s are COW/reference counted (an unfortunate design choice in my opinion). But if you can avoid it - why do reference counting?